ietf-layer0-types.yang

```diff
@@ -44,30 +44,20 @@ module ietf-layer0-types {
      This version of this YANG module is part of RFC XXXX; see
      the RFC itself for full legal notices.";
 
-  revision "2020-01-27" {
+  revision "2020-04-15" {
     description
       "Initial Version";
     reference
       "RFC XXXX: A YANG Data Model for Layer 0 Types";
   }
 
-  typedef operational-mode {
-    type string;
-    description
-      "Vendor-specific mode that guarantees interoperability.
-       It must be an string with the following format:
-       B-DScW-ytz(v) where all these attributes are conforming
-       to the ITU-T recommendation";
-    reference "ITU-T G.698.2 (11/2018) Section 5.3";
-  }
-
   typedef dwdm-n {
     type int16;
     description
       "The given value 'N' is used to determine the nominal
       central frequency.
       The nominal central frequency, 'f' is defined by,
-      f = 193100.000 GHz + N x 12.500 GHz,
+      f = 193100.000 GHz + N x channel-spacing (measured in GHz),
```

Comment:
I would prefer the word **"nominal central frequency granularity**" instead of "channel spacing" following RFC7698 definitions.

Reply:
> I would prefer the word **"nominal central frequency granularity**" instead of "channel spacing" following RFC7698 definitions.

I think the nominal central frequency granularity is not the right term here, and insist on using "channel spacing". RFC7698 is referencing G.694.1, and I paste the following text from G.694.1: 

Section 6 for WSON: 

_- For channel spacings of 12.5 GHz on a fibre, the allowed channel frequencies (in THz) are defined by:   193.1 + n × 0.0125 where n is a positive or negative integer including 0_ 
_- For channel spacings of 25 GHz on a fibre, the allowed channel frequencies (in THz) are defined by:   193.1 + n × 0.025 where n is a positive or negative integer including 0_ 
_- For channel spacings of 50 GHz on a fibre, the allowed channel frequencies (in THz) are defined by:   193.1 + n × 0.05 where n is a positive or negative integer including 0_ 
_- For channel spacings of 100 GHz or more on a fibre, the allowed channel frequencies (in THz) are defined by:   193.1 + n × 0.1 where n is a positive or negative integer including 0_ 

Section 7 for Flexi-grid:

_- For the flexible DWDM grid, the allowed frequency slots have a nominal central frequency (in THz) defined by:   193.1 + n × 0.00625 where n is a positive or negative integer including 0  and 0.00625 is the nominal central frequency granularity in THz_ 

From the above we can see the channel spacing have a set of value and can be different according to deployment. However the term norminal central frequency granularity is only applicable to flexi-grid and have only one value as 6.25GHz. Therefore using channel spacing here is more accurate to capture what is specified in G.694.1.

Reply:
I have checked G.694.1 and I can confirm that the term channel spacing is the right term defined in ITU-T for the 12.5 GHz/25 GHz/50 GHz/100 GHz fixed grids.

However, in case of flexible grid, the term nominal central frequency granularity is used, in clause 7, to describe the 6.25 GHz granularity of the flexible grid definition and in Appendix I.2 to describe the as well as any implementation constraints:

> For example, an application could be defined where the nominal central frequency granularity is 12.5 GHz (by only requiring values of n that are even) and that only requires slot widths as a multiple of 25 GHz (by only requiring values of m that are even).

A similar text is also reported in section 3.4.1 of RFC7698:

>    As stated in Clause 7 of [G.694.1], the flexible DWDM grid has a
>    nominal central frequency granularity of 6.25 GHz and a slot width
>    granularity of 12.5 GHz.  However, devices or applications that make
>    use of the flexible grid might not be capable of supporting every
>    possible slot width or position.  In other words, applications may be
>    defined where only a subset of the possible slot widths and positions
>    is required to be supported.  For example, an application could be
>    defined where the nominal central frequency granularity is 12.5 GHz
>    (by only requiring values of n that are even) and where slot widths
>    are a multiple of 25 GHz (by only requiring values of m that are
>    even).
> 

From a YANG modelling perspective we need two different attributes to represent the 6,25 GHz and the implementation constraints (e.g., the 12.5 GHz)

RFC6205 re-uses the term channel spacing (C.S.) to represent the 6.25GHz and RFC8363 defines the nominal central frequency granularity as:

>    C.F.G (Central Frequency Granularity, 8 bits): A positive integer.
>    Its value indicates the multiple of C.S., in terms of central
>    frequency granularity.
> 

Based on this analysis, I think we could use the term channel spacing to represent the 6.25GHz; redefine the nominal-central-frequency-granularity attribute as an uint8 (in alignment with RFC8363) and move this attribute within the flexi-grid-label-step grouping

@haomianzheng @sergiobelotti : what do you think?

Reply:
Looking at the many different groupings defined for DWDM fixed and flexible grid according G.694.1, I am wondering whether this is the right approach. Flexible grid is a superset of fixed grid and even for flexible grid there can be implementation constraints for n and m values. So, a generalized grouping could be defined for both fixed and flexible grid where n and m constraints are defined and all cases are covered. This would make many different micro groupings obsolete.

Reply:
@italobusi  : 
not clear to me your proposal : what do you want to say with "redefine the nominal-central-frequency-granularity attribute as an uint8 (in alignment with RFC8363) and move this attribute within the flexi-grid-label-step grouping". 
Btw I think we should avoid to go on with small separate grouping between flex grid and fixed grid : we are not talking about two different things but one appear to me a superset of the other so we should try as much as possible to have a common structure to represent both and only , when there is not alternative, using specific grouping.

@italobusi @haomianzheng 
I think this discussion about the right definition with respect "granularity" in the nominal central frequency context, is a further example of the confusion , unfortunately, inside the WSON, FlexGrid model. If you take e.g. RFC 7699  it is even using internal at the same section 2 about "Overview of Flexi-Grid" the two terms mixing them
" The basis of the work is to allow a data channel to be formed
   from an abstract grid anchored at 193.1 THz and selected on a channel
   spacing of 6.25 GHz with a variable slot width measured in units of
   12.5 GHz.  Individual allocations may be made on this basis from
   anywhere in the spectrum, subject to allocations not overlapping.

   [G.694.1] provides clear guidance on the support of flexible grid by
   implementations in Section 2 of Appendix I:

      The flexible DWDM grid defined in clause 7 has a nominal central
      frequency granularity of 6.25 GHz and a slot width granularity of
      12.5 GHz.  However, devices or applications that make use of the
      flexible grid may not have to be capable of supporting every
      possible slot width or position.  In other words, applications may
      be defined where only a subset of the possible slot widths and
      positions are required to be supported.

      For example, an application could be defined where the nominal
      central frequency granularity is 12.5 GHz (by only requiring
      values of n that are even) and that only requires slot widths as a
      multiple of 25 GHz (by only requiring values of m that are even)."
and then suggesting to look at RFC 7698...
"Some additional background on the use of GMPLS for flexible grids can
   be found in [RFC7698].".
Now, frankly speaking I do not care what terms your prefer to use, obviously both are fine, my suggestion was only related to a couple of considerations:
1) trying to avoid proliferation of different definitions 
2) I consider RFC 7698 one of the best L0 document written in IETF and referencing that surely could be an assurance of avoiding misunderstanding.
The fact that "actually" nominal central frequency granularity/channel spacing in flex grid is 6.25 does not mean that it could be different in short time adding new values (as required also in IETF) , so in my view it should not associate different definitions (nsfg vs. cs) to the technology (flexible vs. fixed).